src/manager.ts

```diff
@@ -218,4 +219,35 @@ export class Manager {
         this.extension.completer.command.getCommandsTeX(filePath)
         this.extension.completer.reference.getReferencesTeX(filePath)
     }
+
+    findBinary(config: string, candidates: string[]) {
+        const configuration = vscode.workspace.getConfiguration('latex-workshop')
+        const configCommand = configuration.get(config) as string
+        if (fs.existsSync(configCommand)) {
+            return
+        }
+        hasbin(configCommand, result => {
+            if (result) {
+                return
+            }
+            let found = false
+            for (const candidate of candidates) {
+                if (fs.existsSync(candidate)) {
+                    found = true
+                    vscode.window.showWarningMessage(`'latex-workshop.${config}' value '${configCommand}' is invalid. \
+                                                      LaTeX Workshop found a possible executable file at ${candidate}. \
+                                                      Do you want to set it as 'latex-workshop.${config}'?`,
+                                                     'Yes').then(msg => {
+                        if (msg === 'Yes') {
+                            configuration.update(config, candidate, true)
```

Comment:
* from above, does this update `workspace` or `user` settings? If so should be made clear to user in messages.